packages/crypto-wallet-core/src/transactions/index.ts

```diff
@@ -0,0 +1,29 @@
+import { BTCTxProvider } from './btc';
+import { BCHTxProvider } from './bch';
+import { ETHTxProvider } from './eth';
+
+const providers = {
+  BTC: new BTCTxProvider(),
+  BCH: new BCHTxProvider(),
+  ETH: new ETHTxProvider()
+};
+
+export class TxProvider {
+  get({ chain }) {
+    return providers[chain];
+  }
+
+  create(params) {
+    return this.get(params).create(params);
+  }
+
+  async sign(params): Promise<any> {
+    return this.get(params).sign(params);
+  }
+
+  getSigningAddresses(params) {
```

Comment:
I don't think this belongs on the top level thing, since it's really bitcoin specific